en/components/grids_templates/remote_data_operations.md

```diff
@@ -110,6 +110,68 @@ When requesting data, you need to utilize the [`IForOfState`]({environment:angul
 <button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="grid-sample-4-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
 </div>
 
+
+#### Infinite Scroll
+
+Stepping on the `IForOfState` interface, you could easily adjust your code to achieve an infinite scroll functionality.   
+
+To implement infinite scroll, you'd have to fetch the data page-by-page, i.e. not fetch all the data at once but rather in smaller pages, encompassing data to populate the current grid data view. The data, that's already fetched, will have to be stored in a data structure and you will need to determine the length of a chunk (page) and how many pages there are, i.e. how many records you have. To obtain the infinite scroll functionality, you will also have to keep a track of the last visible data row index in the grid. In this way, using the `startIndex` and `chunkSize` properties, you will know whether the user has scrolled upwards and you'll need to show them an already fetched data from the cached data structure, or will have to request for another data chunk.
+
+To begin, you'd have to use the `ngAfterViewInit` lifecycle hook to fetch the first chunk of data. There, you will need to set the `totalItemCount` property in order to resize the scrollbar accordingly.
+
+```typescript
+public ngAfterViewInit() {
+    ...
+    this._remoteService.loadDataForPage(this.page, this.pageSize, (request) => {
+        if (request.data) {
+            this.grid.totalItemCount = this.page * this.pageSize;
+            this.grid.data = this._remoteService.getCachedData({startIndex: 0, chunkSize: 10});
+            this.totalItems = request.data["@odata.count"];
+            this.totalPageCount = Math.ceil(this.totalItems / this.pageSize);
+            this.grid.isLoading = false;
+        }
+    });
+    ...
+}
+```
+
+After that, you will need to bind yourself to the `onDataPreLoad` output in order to make an appropriate request for the current virtualization state. In the event handler, you will basically have to determine whether to make a new request to the data source, or return data from the cached data structure.
```

Comment:
```suggestion
Additionally, you have to subscribe to the `onDataPreLoad` output so that you can provide the data needed by the grid when it tries to display a different chunk than the currently loaded one. In the event handler you have to determine whether to fetch new data or return data already cached locally.
```